File: tp1/Palindromo.java
class Palindromo  {  
    public static void main(String args[])  { 
        for(;;){ 
            String original, reverse = "";
            original = MyIO.readLine();
            if(original.equals("FIM")){
                System.exit(0);
            }   
            int length = original.length();   
            for ( int i = length - 1; i >= 0; i-- )  
                reverse = reverse + original.charAt(i);  
            if (original.equals(reverse))  
                System.out.println("SIM");  
            else  
                System.out.println("NAO");
        }
   }  
}  

File: tp1/Palindromo.cpp
#include <iostream>
#include <cstring>
using namespace std;

int main(){
    char string1[20];
    int i, length;
    
    for(;;){
        int flag = 0;
        cin >> string1;
        int c = strcmp(string1, "FIM");
        if(c == 0){
            exit(0);
        }
        length = strlen(string1);
        
        for(i=0;i < length ;i++){
            if(string1[i] != string1[length-i-1]){
                flag = 1;
                break;
            }
        }

        if (flag) {
            cout <<"NAO"<< endl; 
        }    
        else {
            cout <<"SIM"<< endl; 
        }
    }

    return 0;
}